src/ORM/DataObject.php

```diff
@@ -3827,7 +3837,7 @@ public function requireDefaultRecords()
         $defaultRecords = $this->config()->uninherited('default_records');
 
         if (!empty($defaultRecords)) {
-            $hasData = DataObject::get_one(static::class);
+            $hasData = DataObject::get(static::class)->setUseCache(true)->count() > 0;
```

Comment:
We don't actually need a record here - we just need to see if there _are_ any records for this class. This is a more performant way to check that.
Probably doesn't actually need the caching here.